OpenRA.Mods.Common/Traits/AttackMove.cs

```diff
@@ -124,9 +105,10 @@ public void ResolveOrder(Actor self, Order order)
 				if (!Info.MoveIntoShroud && !self.Owner.Shroud.IsExplored(cell))
 					return;
 
-				TargetLocation = move.NearestMoveableCell(cell);
-				self.SetTargetLine(Target.FromCell(self.World, TargetLocation.Value), Color.Red);
-				Activate(self, order.OrderString == "AssaultMove");
+				var targetLocation = move.NearestMoveableCell(cell);
+				self.SetTargetLine(Target.FromCell(self.World, targetLocation), Color.Red);
+				assaultMoving = order.OrderString == "AssaultMove";
```

Comment:
I suspect this will activate the `AssaultMoveScanCondition` prematurely if you queue an assault move after an existing attack move? It would be better to pass this into the activity, and then pull it out with `((AttackMoveActivity)activity).IsAssaultMove` in `Tick`.

Reply:
An option for the future, once we are sure that activities that have started are guaranteed to have a chance to clean themselves up, is to move the condition granting/revoking into `AttackMoveActivity.OnFirstRun` and `AttackMoveActivity.OnLastRun`.

I don't think we're quite there yet?

Reply:
Okay. I've passed it into the activity.

> I don't think we're quite there yet?

I'd rather make sure that all attack variations properly work with childactivities first before doing such a thing. I will look into this when converting AttackMove to use childactivities later.